datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -321,23 +321,30 @@ public void start() {
         _heartbeatPeriod.toMillis() * 3, _heartbeatPeriod.toMillis(), TimeUnit.MILLISECONDS);
   }
 
-  private synchronized void createEventThread() {
+  protected synchronized void createEventThread() {
     _eventThread = new CoordinatorEventProcessor();
     _eventThread.setDaemon(true);
   }
 
   private synchronized void startEventThread() {
     if (!_shutdown) {
-      _eventThread.start();
+      CoordinatorEventProcessor eventThread = getEventThread();
+      eventThread.start();
     }
   }
 
   private synchronized boolean stopEventThread() {
     // interrupt the thread if it's not gracefully shutdown
-    while (_eventThread.isAlive()) {
+    CoordinatorEventProcessor eventThread = getEventThread();
```

Comment:
what are we not pointing at _eventThread instead of calling getEventThread() ?

Reply:
Using an API makes it much better to test it with overrides instead of mocking the var, hence I changed the references to use the APIs to fetch the variable value.

Reply:
nice!